waspc/cli/src/Wasp/Cli/Command/CreateNewProject.hs

```diff
@@ -18,14 +18,16 @@ import Wasp.Cli.Command (Command, CommandError (..))
 import qualified Wasp.Cli.Command.Common as Command.Common
 import qualified Wasp.Cli.Common as Common
 import qualified Wasp.Data
-import Wasp.Util (indent)
+import Wasp.Util (indent, projectNameToAppIdentifier)
 import qualified Wasp.Util.Terminal as Term
 
 newtype ProjectName = ProjectName {_projectName :: String}
 
+newtype AppIdentifier = AppIdentifier {_appIdentifier :: String}
+
```

Comment:
@Martinsos, this one's for you.

I might be missing something here, but it seems like we don't need the `_appIdentifer` part (the same goes for `_projectName`). We could simply say `newtype AppIdentifier = AppIdentifier String`, right?

Furthermore, why are we using `newtype`s anyway? We neither define class instances on it nor do we use any smart constructors. If we only wanted a different name for a String, wouldn't it be more appropriate to use a type synonym?

Reply:
Speaking of smart constructors, if we decide to keep the `newtype`, we could create the following function:
```haskell
getAppIdentifier :: String -> Maybe AppIdentifier
```
The function would internally transform the given string from kebab-case to camelCase and verify the result is a valid Wasp identifier (or return `Nothing`).

Reply:
The purpose of `newtype` is to provide some extra safety in the types -> so if a function takes ProjectName, you can't just pass any String. I believe this was the purpose here. I personally am not a big fan of Strings that just go around, if they have any more semantics then just a String. It is better to use type alias as you said. But it can be even better, if it is practical, to use newtype, since that gives us compile time safety.
I believe I was writing this code relatively soon after reading https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/ hehe so maybe I was a bit trigger-happy, but still, I like newtypes for this kind of stuff.

Maybe the nicest thing to do here would be to introduce function called
`parseWaspProjectName :: String -> Either String ProjectName`, where `String` is error message (ah irony, I just used `String` that has semantic meaning without type aliasing -> so even better would be to maybe use type alias for it to call it `ErrorMessage` or smth) and then such function would check that name is valid and it can be parsed into wasp identifier.

We would also adjust `ProjectName` to either contain additional field called `_projectWaspIdentifier`, or we would write function `projectNameToWaspIdentifier :: ProjectName -> String` (again, String could be aliased as Identifier, I think we even have that type alias somewhere) and then use that function on the fly when we need the identifier.
I am ok with both approaches really, not sure what is better!

Therefore, following what I described above, I would remove AppName data type.

Reply:
You don't have to read/understand everything Martin and I talked about here, it's enough to read [the conclusion](https://github.com/wasp-lang/wasp/pull/700#discussion_r955900024).

Reply:
Got it - I followed the conclusion :+1: